improver/utilities/temporal.py

```diff
@@ -254,3 +266,217 @@ def extract_nearest_time_point(cube, dt, time_name="time", allowed_dt_difference
+        )
+
+        # Create a local time coordinate to help with plotting data.
+        local_time_coord_standards = TIME_COORDS["time_in_local_timezone"]
+        local_time_units = cf_units.Unit(
+            local_time_coord_standards.units,
+            calendar=local_time_coord_standards.calendar,
+        )
+        timezone_points = np.array(
+            np.round(local_time_units.date2num(local_time)),
+            dtype=local_time_coord_standards.dtype,
+        )
+        output_cube.add_aux_coord(
+            AuxCoord(
+                timezone_points,
+                long_name="time_in_local_timezone",
+                units=local_time_units,
+            )
+        )
+        output_cube.add_aux_coord(
+            AuxCoord(
+                self.time_points,
+                bounds=self.time_bounds,
+                standard_name="time",
+                units=self.time_units,
+            ),
+            (0, 1),
+        )
+        return output_cube
+
+    def fill_timezones(self, input_cube):
+        """
+        Populates the output cube data with data from input_cube. This is done by
+        multiplying the inverse of the timezone_cube.data with the input_cube.data and
+        summing along the time axis. Because timezone_cube.data is a mask of 1 and 0,
+        inverting it gives 1 where we WANT data and 0 where we don't. Summing these up
+        produces the result. The same logic can be used for times.
+        Modifies self.output_cube and self.time_points.
+
+        Args:
+            input_cube (iris.cube.Cube):
+                Cube of data to extract timezone-offsets from. Must contain a time
+                coord spanning all the timezones.
+
+        Raises:
+            TypeError:
+                If combining the timezone_cube and input_cube results in float64 data.
+                (Hint: timezone_cube should be int8 and input cube should be float32)
+        """
+        # Get the output_data
+        result = input_cube.data * (1 - self.timezone_cube.data)
+        self.output_data = result.sum(axis=0)
+
+        # Check resulting dtype
+        enforce_dtype("multiply", [input_cube, self.timezone_cube], result)
+
+        # Sort out the time points
+        input_time_points = input_cube.coord("time").points
+        # Add scalar coords to allow broadcast to spatial coords.
+        times = input_time_points.reshape((len(input_time_points), 1, 1)) * (
+            1 - self.timezone_cube.data
+        )
+        self.time_points = times.sum(axis=0)
+
+        # Sort out the time bounds (if present)
+        bounds_offsets = self.get_time_bounds_offset(input_cube)
+        if bounds_offsets is not None:
+            # Add scalar coords to allow broadcast to spatial coords.
+            self.time_bounds = bounds_offsets.reshape(
+                (1, 1, 2)
+            ) + self.time_points.reshape(list(self.time_points.shape) + [1])
+
+    @staticmethod
+    def get_time_bounds_offset(input_cube):
```

Comment:
This function is not currently explicitly unit tested. If you think you have sufficient coverage under the process tests, this should be made a private method.

Reply:
Ok. Leading underscore added.